src/osx/cocoa/window.mm

```diff
@@ -2568,6 +2580,53 @@ void wxOSXCocoaClassAddWXMethods(Class c, wxOSXSkipOverrides skipFlags)
     wxCocoaGestures::EraseForObject(this);
 }
 
+void wxWidgetCocoaImpl::BeginNativeKeyDownEvent( NSEvent* event )
+{
+    m_lastKeyDownEvent = event;
+    m_lastKeyDownWXSent = false;
+}
+
+void wxWidgetCocoaImpl::EndNativeKeyDownEvent()
+{
+    m_lastKeyDownEvent = nil;
+    m_lastKeyDownWXSent = false;
+}
+
+bool wxWidgetCocoaImpl::IsInNativeKeyDown()
+{
+    return m_lastKeyDownEvent != nil;
+}
+
+NSEvent* wxWidgetCocoaImpl::GetLastNativeKeyDownEvent()
+{
+    return m_lastKeyDownEvent;
```

Comment:
Should we assert here that it's not `nil`? Otherwise I don't really see much point in having this (and `IsInNativeKeyDown()`) function(s), IMO the code using them is not really any better than the old code which used `m_lastKeyDownEvent` directly.

Reply:
I preferred having a clear wording of the methods and their intention instead of a minimal interface. I'll add an assert as well.